src/logic/hooks/useEstimateSafeTxGas.tsx

```diff
@@ -0,0 +1,49 @@
+import { Operation } from '@gnosis.pm/safe-react-gateway-sdk'
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+
+import { estimateSafeTxGas } from 'src/logic/safe/transactions/gas'
+import { currentSafe } from 'src/logic/safe/store/selectors'
+
+type UseEstimateSafeTxGasProps = {
+  isCreation: boolean
+  txData: string
+  txRecipient: string
+  txAmount: string
+  operation?: Operation
+}
+
+export const useEstimateSafeTxGas = ({
+  isCreation,
+  txData,
+  txRecipient,
+  txAmount,
+  operation,
+}: UseEstimateSafeTxGasProps): string => {
+  const [safeTxGasEstimation, setSafeTxGasEstimation] = useState<string>('0')
+  const { address: safeAddress, currentVersion: safeVersion } = useSelector(currentSafe) ?? {}
+
+  useEffect(() => {
+    if (!isCreation) return
+    const estimateSafeTxGasCall = async () => {
+      try {
+        const safeTxGasEstimation = await estimateSafeTxGas(
+          {
+            safeAddress,
+            txData,
+            txRecipient,
+            txAmount: txAmount || '0',
```

Comment:
For future reference: we have 'default' assignments for the `value`, `operation`, etc. across the Safe (especially in `create`/`processTransaction`. Maybe we should consider centralising these for all of the default assignments.